cmd/root.go

```diff
@@ -67,100 +26,29 @@ const commentDateFormat = "15:04 PM, January 2 2006"
 func Execute() {
+	// Create a temporary logger that we can use if an error occurs before the real one is instantiated
+	log := logrus.New()
 	if err := RootCmd.Execute(); err != nil {
 		log.Fatal(err)
 	}
 }
 
-// getErrorBody reads the HTTP response body of a JIRA API response,
-// logs it as an error, and returns an error object with the contents
-// of the body. If an error occurs during reading, that error is
-// instead printed and returned. This function closes the body for
-// further reading.
-func getErrorBody(res *jira.Response) error {
-	defer res.Body.Close()
-	body, err := ioutil.ReadAll(res.Body)
-	if err != nil {
-		log.Errorf("Error occured trying to read error body: %v", err)
-		return err
-	} else {
-		log.Debugf("Error body: %s", body)
-		return errors.New(string(body))
-	}
-}
-
-// makeGHRequest takes an API function from the GitHub library
-// and calls it with exponential backoff. If the function succeeds, it
-// stores the value in the ret parameter, and returns the HTTP response
-// from the function, and a nil error. If it continues to fail until
-// a maximum time is reached, the ret parameter is returned as is, and a
-// nil HTTP response and a timeout error are returned.
-//
-// It is nearly identical to makeJIRARequest, but returns a GitHub API response.
-func makeGHRequest(f func() (interface{}, *github.Response, error)) (interface{}, *github.Response, error) {
-	var ret interface{}
-	var res *github.Response
-	var err error
-
-	op := func() error {
-		ret, res, err = f()
-		return err
-	}
-
-	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = rootCmdCfg.GetDuration("timeout")
-
-	er := backoff.Retry(op, b)
-	if er != nil {
-		return nil, nil, er
-	}
-
-	return ret, res, err
-}
-
-// makeJIRARequest takes an API function from the JIRA library
-// and calls it with exponential backoff. If the function succeeds, it
-// stores the value in the ret parameter, and returns the HTTP response
-// from the function, and a nil error. If it continues to fail until
-// a maximum time is reached, the ret parameter is returned as is, and a
-// nil HTTP response and a timeout error are returned.
-//
-// It is nearly identical to makeGHRequest, but returns a JIRA API response.
-func makeJIRARequest(f func() (interface{}, *jira.Response, error)) (interface{}, *jira.Response, error) {
-	var ret interface{}
-	var res *jira.Response
-	var err error
-
-	op := func() error {
-		ret, res, err = f()
-		return err
-	}
-
-	b := backoff.NewExponentialBackOff()
-	b.MaxElapsedTime = rootCmdCfg.GetDuration("timeout")
-
-	er := backoff.Retry(op, b)
-	if er != nil {
-		return nil, nil, er
-	}
-
-	return ret, res, err
-}
-
 // getGitHubClient initializes a GitHub API client with an OAuth client for authentication,
 // then makes an API request to confirm that the service is running and the auth token
 // is valid
-func getGitHubClient(token string) (*github.Client, error) {
+func getGitHubClient(config lib.Config) (*github.Client, error) {
```

Comment:
this should be part of the the clients file. lib should know how to create a new github client given a config.